### PR TITLE
Add the ability to specify padding for elementHandle.screenshot()

### PR DESCRIPTION
**Pull request by Glennvd**
Proposal to add an easy option to add padding to the elementHandle.screenshot() function. I'm not quite sure yet how the api interface should look, but this is a first working proposal.

### CODE REVIEW

**Comment by Glennvd** [comment]
I signed it (covered under corporate agreement)

File: lib/ElementHandle.js [hunk 1]
```diff
@@ -152,6 +152,37 @@ class ElementHandle extends JSHandle {
...
+
+    if (options.padding) {
+      if (options.padding.constructor !== Array || options.padding.length > 4)
+          throw Error('Invalid padding parameter (should be [top, right, bottom, left])');     
+      switch (options.padding.length){
+        case 1:
+          clip.x -= options.padding[0];
+          clip.y -= options.padding[0];
+          clip.width += options.padding[0];
+          clip.height += options.padding[0];
```
Comment (JoelEinbinder) [inline lib/ElementHandle.js]: Shouldn't this be:
`clip.height += options.padding[0] + options.padding[0]`?
A 2x2 square with a padding of 3 should be an 8x8 square.

**Comment by JoelEinbinder** [comment]
Thanks for the PR! What is your use case for this? It seems like it could be done currently by using `element.boundingBox()` and `page.screenshot()`. I am not sure if it is worth it to add a shortcut to `element.screenshot()` for this.

**Comment by aslushnikov** [comment]
This looks easily polyfillable to me. Please feel free to object if there are good reasons why this should be a part of the api.

**Pull request closed without merging**